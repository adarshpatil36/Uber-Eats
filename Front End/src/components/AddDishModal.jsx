import React from "react";

import { Modal, Button, Form, Row, Col } from "react-bootstrap";

export default function AddDishModal({
  handleClose,
  show,
  handleNewDish,
  saveNewDish,
  newDish,
}) {
  return (
    <>
      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Add Dish</Modal.Title>
        </Modal.Header>
        <Form className="modal-form">
          <Form.Group
            as={Row}
            className="mb-6 m-3"
            controlId="formPlaintextEmail"
          >
            <Form.Label column sm="3">
              Dish Name
            </Form.Label>
            <Col sm="10">
              <Form.Control
                type="text"
                id="name"
                className="modal-Qty"
                value={newDish["name"]}
                onChange={(e) => {
                  handleNewDish(e);
                }}
              />
            </Col>
            <Form.Label column sm="3">
              Description
            </Form.Label>
            <Col sm="10">
              <Form.Control
                type="text"
                id="description"
                className="modal-Qty"
                onChange={(e) => {
                  handleNewDish(e);
                }}
              />
            </Col>
            <Form.Label column sm="3">
              Price
            </Form.Label>
            <Col sm="10">
              <Form.Control
                type="text"
                id="price"
                className="modal-Qty"
                onChange={(e) => {
                  handleNewDish(e);
                }}
              />
            </Col>
            <Form.Label column sm="3">
              Rating
            </Form.Label>
            <Col sm="10">
              <Form.Control
                type="text"
                id="rating"
                className="modal-Qty"
                onChange={(e) => {
                  handleNewDish(e);
                }}
              />
            </Col>
          </Form.Group>
        </Form>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Close
          </Button>
          <Button variant="primary" onClick={saveNewDish}>
            Save Dish
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}
