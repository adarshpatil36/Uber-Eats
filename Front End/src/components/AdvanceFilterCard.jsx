import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Accordion from "react-bootstrap/Accordion";

export const AdvanceFilterCard = (props) => {
  return (
    <div className="advanceFilterCard">
      <p>All Stores</p>
      <Accordion defaultActiveKey="0">
        <Accordion.Item eventKey="0">
          <Accordion.Header>Dietary</Accordion.Header>
          <Accordion.Body>
            <div>
              <input type="radio" id="age1" value="30" />
              <span>Vegeterian</span>
            </div>
            <div>
              <input type="radio" id="age1" value="30" />
              <span>Non Vegeterian </span>
            </div>
          </Accordion.Body>
        </Accordion.Item>
        <Accordion.Item eventKey="1">
          <Accordion.Header>Price Range</Accordion.Header>
          <Accordion.Body className="priceRange">
            <button>$</button>
            <button>$$</button>
            <button>$$$</button>
            <button>$$$$</button>
          </Accordion.Body>
        </Accordion.Item>
        <Accordion.Item eventKey="2">
          <Accordion.Header>Max Delivery Fee</Accordion.Header>
          <Accordion.Body>
            <input
              type="range"
              className="form-range"
              min="0"
              max="5"
              step="1"
              id="customRange3"
            />
            <div className="range">
              <p>$1</p>
              <p>$3</p>
              <p>$5</p>
              <p>$5+</p>
            </div>
          </Accordion.Body>
        </Accordion.Item>
        <Accordion.Item eventKey="3">
          <Accordion.Header>Sort</Accordion.Header>
          <Accordion.Body>
            <button>Picked for you</button>
            <button>Most Popular</button>
            <button>Rating</button>
            <button>Delivery Time</button>
            <button>Allergy Friendly</button>
          </Accordion.Body>
        </Accordion.Item>
      </Accordion>
    </div>
  );
};

AdvanceFilterCard.propTypes = {
  props: PropTypes,
};

const mapStateToProps = (state) => ({});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(AdvanceFilterCard);
