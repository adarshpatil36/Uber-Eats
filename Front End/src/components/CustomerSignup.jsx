import React, { useState } from "react";
import PropTypes from "prop-types";
import { Form, Button } from "react-bootstrap";

import Logo from "./Logo";
import { useHistory } from "react-router";
import axios from "axios";
import { CONSTANTS } from "../constants/constants";
import { ACTION_TYPE } from "../actions/ActionTypes";
import { connect } from "react-redux";

const CustomerSignup = ({ changeActiveTab, loginUser }) => {
  const [data, setData] = useState({
    firstName: "",
    lastName: "",
    user: "",
    address: "",
    email: "",
    password: "",
    contact: "",
    country: "",
  });
  let history = useHistory();

  const loginTab = () => {
    history.push("./login");
  };
  const handleChange = (e) => {
    const { id, value } = e.target;
    setData({ ...data, [id]: value });
  };

  const signUp = async () => {
    const { confPassword, ...postData } = data;
    await axios.post("http://localhost:8080/users", postData).then((res) => {
      if (res.status === 200) {
        postData["id"] = res.data.id;
        loginUser(postData);
        changeActiveTab(CONSTANTS.DASHBOARD);
        history.push("/dashboard");
      } else {
        console.log("Data post failed ");
      }
    });
  };
  const redirectToLogin = () => {
    history.push("/login");
  };

  return (
    <div className="login">
      <Logo navigateTo={redirectToLogin} />
      <div className="loginTab">
        <h5>Let's get started </h5>
        <Form>
          <Form.Group className="mb-8">
            <Form.Label>First Name</Form.Label>
            <Form.Control
              id="firstName"
              placeholder="First Name"
              value={data["firstName"]}
              onChange={handleChange}
              required
            />
          </Form.Group>

          <Form.Group className="mb-8">
            <Form.Label>Last Name</Form.Label>
            <Form.Control
              id="lastName"
              placeholder="Last Name"
              value={data["lastName"]}
              onChange={handleChange}
              required
            />
          </Form.Group>

          <Form.Group className="mb-8">
            <Form.Label>User Name</Form.Label>
            <Form.Control
              id="user"
              placeholder="User Name"
              value={data["user"]}
              onChange={handleChange}
              required
            />
          </Form.Group>

          <Form.Group className="mb-8">
            <Form.Label>Address</Form.Label>
            <Form.Control
              id="address"
              placeholder="Address"
              value={data["address"]}
              onChange={handleChange}
              required
            />
          </Form.Group>

          <Form.Group className="mb-8">
            <Form.Label>Country</Form.Label>
            <Form.Control
              id="country"
              placeholder="Country"
              value={data["country"]}
              onChange={handleChange}
              required
            />
          </Form.Group>

          <Form.Group className="mb-8">
            <Form.Label>Mobile Number</Form.Label>
            <Form.Control
              id="contact"
              placeholder="Mobile Number"
              value={data["contact"]}
              onChange={handleChange}
              required
            />
          </Form.Group>

          <Form.Group className="mb-8" controlId="formBasicEmail">
            <Form.Label>Email address</Form.Label>
            <Form.Control
              id="email"
              placeholder="Email ID"
              value={data["email"]}
              onChange={handleChange}
              type="email"
              required
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="formBasicPassword">
            <Form.Label>Password</Form.Label>
            <Form.Control
              type="password"
              id="password"
              placeholder="Password"
              value={data["password"]}
              onChange={handleChange}
              required
            />
          </Form.Group>

          <Form.Group controlId="formFileSm" className="mb-3">
            <Form.Label>Profile Picture</Form.Label>
            <Form.Control type="file" size="sm" />
          </Form.Group>
          <Button variant="primary" onClick={signUp}>
            Next
          </Button>
          <Form.Group controlId="formFileSm" className="mb-3 bottomGroup">
            <p>
              Already user?
              <span onClick={() => loginTab()}> Sign in</span>
            </p>
          </Form.Group>
        </Form>
      </div>
    </div>
  );
};

CustomerSignup.propTypes = {
  props: PropTypes,
};

const mapStateToProps = (state) => ({
  activeTab: state.data.activeTab,
  isRestaurant: state.data.activeTab === CONSTANTS.RESTAURANT_SIGNUP,
});

const mapDispatchToProps = (dispatch) => {
  return {
    changeActiveTab: (tabID) =>
      dispatch({ type: ACTION_TYPE.SET_ACTIVE_TAB, value: tabID }),
    loginUser: (data) =>
      dispatch({ type: ACTION_TYPE.SET_LOGIN_DATA, value: data }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CustomerSignup);
