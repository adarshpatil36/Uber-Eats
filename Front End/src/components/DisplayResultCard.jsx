import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import { ACTION_TYPE } from "../actions/ActionTypes";
import { URLS } from "../constants/api";
import { restaurants } from "../constants/static data/restaurants";

const DisplayResultCard = ({ setSelectedRestaurant }) => {
  let history = useHistory();

  const openRestaurantCard = (res) => {
    setSelectedRestaurant(res);
    history.push("/restaurant");
  };

  return (
    <div className="displayResultCard">
      <h3> Popular near you</h3>
      <div class="container">
        <div class="flex">
          {restaurants.map((item) => (
            <div
              class="card"
              id={item.id}
              onClick={() => openRestaurantCard(item)}
            >
              <img src={item.url} alt="" />
              <div class="card-body">
                <h3>{item.title}</h3>
                <p>{item.deliveryFee}</p>
                <p>Delivery Time: {item.deliveryTime}</p>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

DisplayResultCard.propTypes = {
  props: PropTypes,
};

const mapStateToProps = (state) => ({});

const mapDispatchToProps = (dispatch) => {
  return {
    setSelectedRestaurant: (res) =>
      dispatch({ type: ACTION_TYPE.SET_SELECTED_RESTAURANT, value: res }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(DisplayResultCard);
