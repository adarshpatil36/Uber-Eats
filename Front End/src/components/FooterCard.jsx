import React from "react";

export default function FooterCard() {
  return (
    <footer className="footerCard">
      <span>Â© 2020 Uber Technologies, Inc.</span>
      <span>Privacy Policy | Terms of Use</span>
    </footer>
  );
}
