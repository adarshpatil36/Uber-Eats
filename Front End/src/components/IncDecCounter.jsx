import React from "react";

const IncDecCounter = ({ item, increaseQuantity, decreaseQuantity }) => {
  const increamentCount = () => {
    increaseQuantity(item);
  };
  const decreamentCount = () => {
    if (item.quantity !== 0) {
      decreaseQuantity(item);
    }
  };
  return (
    <span>
      <button className="quantity__minus" onClick={decreamentCount}>
        -
      </button>
      <label
        name="quantity"
        type="text"
        className="quantity__input"
        data-testid="incDecCounter"
      >
        {item.quantity || 0}
      </label>
      <button className="quantity__plus" onClick={increamentCount}>
        +
      </button>
    </span>
  );
};

export default IncDecCounter;
