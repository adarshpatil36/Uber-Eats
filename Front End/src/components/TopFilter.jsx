import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { URLS } from "../constants/api";

export const TopFilter = (props) => {
  return (
    <div className="topFilter">
      {URLS.TOP_FILTER.map((item) => {
        return <img src={item} alt="" onClick={() => alert("Hello")}></img>;
      })}
    </div>
  );
};

TopFilter.propTypes = {
  props: PropTypes,
};

const mapStateToProps = (state) => ({});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(TopFilter);
