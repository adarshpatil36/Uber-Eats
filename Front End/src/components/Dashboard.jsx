import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { TopFilter } from "./TopFilter";
import NavBar from "./NavBar";
import DisplayResultCard from "./DisplayResultCard";
import { FilterCard } from "./FilterCard";
import OfferCard from "./OfferCard";
import SectionCard from "./SectionCard";
import { useLocation } from "react-router";
import ToastCard from "./ToastCard";

export const Dashboard = ({ userData, changeActiveTab }) => {
  const location = useLocation();

  const [showToast, setShowToast] = useState(false);
  useEffect(() => {
    console.log(location.state);
    if (location.state === "orderPlaced") {
      window.scrollTo(0, 0);
      setShowToast(true);
    }
  }, []);

  return (
    <div>
      <NavBar />
      {showToast && <ToastCard />}
      <TopFilter />
      <hr />
      <OfferCard />
      <SectionCard></SectionCard>
    </div>
  );
};

Dashboard.propTypes = {
  userData: PropTypes.object,
};

const mapStateToProps = (state) => ({
  userData: state.data.userData,
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
