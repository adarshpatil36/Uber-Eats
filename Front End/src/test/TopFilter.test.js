import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import ReactDOM from "react-dom";

import renderer from "react-test-renderer";
import FooterCard from "../components/FooterCard";
import IncDecCounter from "../components/IncDecCounter";
import { TopFilter } from "../components/TopFilter";

describe("TopFilter testing", () => {
  it("renders TopFilter without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TopFilter></TopFilter>, div);
  });

  it("snapshot test of TopFilter", () => {
    const tree = renderer.create(<TopFilter></TopFilter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
