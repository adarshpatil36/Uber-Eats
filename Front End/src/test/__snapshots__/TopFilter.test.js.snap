// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncDecCounter testing renders correctly 1`] = `
<div
  className="topFilter"
>
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/deals.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/grocery.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/convenience.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/alcohol.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/pharmacy.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/baby.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/specialty_foods.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/flowers.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/top_eats.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/cuisines/asian.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/cuisines/japanese.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/cuisines/healthy.png"
  />
</div>
`;

exports[`TopFilter testing snapshot test of TopFilter 1`] = `
<div
  className="topFilter"
>
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/deals.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/grocery.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/convenience.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/alcohol.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/pharmacy.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/baby.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/specialty_foods.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/flowers.jpg"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/top_eats.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/cuisines/asian.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/cuisines/japanese.png"
  />
  <img
    alt=""
    onClick={[Function]}
    src="https://d4p17acsd5wyj.cloudfront.net/shortcuts/cuisines/healthy.png"
  />
</div>
`;
